lib/core/include/irods/unique_user.hpp

```diff
@@ -0,0 +1,90 @@
+#ifndef IRODS_UNIQUE_USER_HPP
+#define IRODS_UNIQUE_USER_HPP
+
+/// \file
+
+#include "irods/irods_exception.hpp"
+#include "irods/rodsErrorTable.h"
+
+#include <string_view>
+
+namespace irods::experimental
+{
+    /// A class designed to hold a fully-qualified iRODS username.
+    ///
+    /// This class should be used when a fully-qualified iRODS username is required.
+    ///
+    /// Instances of this class are immutable outside of copy and move opertaions.
+    ///
+    /// \since 4.3.1
+    class unique_user
```

Comment:
Is `fq_username` a better name?
How about `unique_username`?
Does `unique_user` convey things clearly?

`user` kind of implies `username` so I think all of these work.

Will think about this a little more.

Reply:
After discussion, we've decided to go with the following.
```cpp
fully_qualified_username fqun{"rods", "tempZone"};
```

Reply:
Updated.